Sources/ForageSDK/Foundation/Telemetry/ForageLogger.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  File.swift
+//
+//
+//  Created by Danilo Joksimovic on 2023-07-26.
+//
+
+import Datadog
+import Foundation
+
+// TODO: move this?
+protocol ForageLogger {
+    func setCustomerID(_ customerID: String)
+    func setMerchantAccount(_ merchantAccount: String)
+    func initialize(_ environment: EnvironmentTarget)
+
+    func notice(_ message: String, attributes: [String: Encodable]?)
+    func info(_ message: String, attributes: [String: Encodable]?)
+    func warn(_ message: String, error: Error?, attributes: [String: Encodable]?)
+    func error(_ message: String, error: Error?, attributes: [String: Encodable]?)
+    func critical(_ message: String, error: Error?, attributes: [String: Encodable]?)
+}
+
+public class DatadogLogger: ForageLogger {
+    static let shared = DatadogLogger()
+    private static let DD_CLIENT_TOKEN: String = "pub1e4572ba0f5e53df108c333d5ec66c02"
+    private static var logger: Logger? = nil
+
+    private init() {}
+
+    internal func initialize(_ environment: EnvironmentTarget) {
+        if DatadogLogger.logger != nil {
+            return
+        }
+
+        // TODO: only initialize if not running locally?
+
+        Datadog.initialize(
+            appContext: .init(),
+            trackingConsent: .granted,
+            configuration: Datadog.Configuration
+                .builderUsing(clientToken: DatadogLogger.DD_CLIENT_TOKEN, environment: environment.rawValue)
+                .set(serviceName: "ios-sdk")
+                .set(endpoint: .us1)
+                .build()
+        )
+
+        Datadog.verbosityLevel = .info
+
+        DatadogLogger.logger = Logger.builder
+            .sendNetworkInfo(true)
+            // how do we decide whether we're in local dev or not?
+            .sendLogsToDatadog(true)
+            .printLogsToConsole(true, usingFormat: .shortWith(prefix: "[forage-ios-sdk] "))
```

Comment:
I decided against enabling this field. I don't want to clog up client logs with a ton of our statements, we just want to be able to emit them to our own datadog instance.

Reply:
657bf39a9c0b05d39ce7e681271fb19e7bd8447c